### ISSUE TITLE
PermissionManager:Define&Load issure

### ISSUE THREAD

**Issue by JhouXerox** [abpframework/abp#9533]
hello,in my way to use abp.
I want to use single Identityserver4 with permissions.

**How should I load permissions from file directory?**
If I have done an application's dev,I can just copy an application.contracts file which include PermissionDefinitionProvider to the directory.
And then let ID4 host to load the file.
Then I can use Volo.Abp.Identity.Web to configure the user/role permission.

Another question is:
 **Is there anyway to define permissions and save them in db? And load them from db not PermissionDefinitionProvider?**

**Comment by JhouXerox** [comment]
Another way,Your DDD E-book is very Inspired.
I like it ,and I'm trying to translate it to chinese.
How can I upload to you after I've done?
Maybe it will be helpful to chinese abp users.

**Comment by maliming** [comment]
Please check the Permission document.

https://docs.abp.io/en/abp/latest/Authorization#permission-system
https://docs.abp.io/en/abp/latest/Modules/Permission-Management

>  I'm trying to translate it to Chinese.

Someone already did this. : )

https://www.cnblogs.com/YGYH/p/14937097.html#_label0

**Comment by JhouXerox** [comment]
wow,Thank you for your prompt reply
I will check the document.(chinese doc about permission is only "TODO." word in the doc.So I missed a lot.Thank you)

**Comment by JhouXerox** [comment]
@maliming hi,I've checked these docs.

According to Authorization about **Defining Permissions**

> ABP automatically discovers this class. No additional configuration required!

**how can I load my assembly which contains PermissionDefinitionProvider and inject to abp?**

**Comment by JhouXerox** [comment]
I want to automaclly add permissions on running time.Not coding time.
I found a function


`Configure<AbpPermissionOptions>(options =>
{
    options.DefinitionProviders.AddIfNotContains(new List<Type>()
    {
        typeof(AdminPermissionDefinitionProvider)
    });
});`

but there's an exception:
ComponentNotRegisteredException: The requested service 'Com.Bull.SmartCanteen.Admin.Permissions.AdminPermissionDefinitionProvider' has not been registered. To avoid this exception, either register a component to provide the service, check for service registration using IsRegistered(), or use the ResolveOptional() method to resolve an optional dependency.


I searched about  IsRegistered and ResolveOptional,but nothing.

Is there anyway I can regist module by reflection at sometime?
then I can load my permissions from file directory.
